chirp/drivers/retevis_ra87.py

```diff
@@ -0,0 +1,1250 @@
+        rs = RadioSettingValueString(0, 16, tmp, False, CHARSET_DTMF_DIGITS)
+        rset = RadioSetting("dtmfe.eot", "EOT PTT-ID", rs)
+        rset.set_apply_callback(apply_dmtf_frame, _dtmfe.eot, 16)
+        dtmf_enc.append(rset)
+
+        # menu 32 - DT.M
+        rs = RadioSettingValueBoolean(_settings.dtm)
+        rset = RadioSetting("dtm", "DTMF Sidetone", rs)
+        dtmf_enc.append(rset)
+
+        # menu 31 - DT.L
+        rs = RadioSettingValueBoolean(_settings.dtl)
+        rset = RadioSetting("dtl", "DTMF Key Lock", rs)
+        rset.set_doc("Menu 31")
+        dtmf_enc.append(rset)
+
+        # menu 29 - DT.H
+        rs = RadioSettingValueBoolean(_settings.dth)
+        rset = RadioSetting("dth", "DTMF Hold", rs)
+        rset.set_doc("Menu 29")
+        dtmf_enc.append(rset)
+
+        # menu 30 - PA
+        options = ["100", "250", "500", "750", "1000", "1500", "2000"]
+        rs = RadioSettingValueList(options, options[_settings.pa])
+        rset = RadioSetting("pa", "DTMF Pause [ms]", rs)
+        rset.set_doc("Menu 30")
+        dtmf_enc.append(rset)
+
+        # menu 28 - SPD
+        options = ["Fast", "Slow"]
+        rs = RadioSettingValueList(options, options[_settings.spd])
+        rset = RadioSetting("spd", "DTMF Speed", rs)
+        rset.set_doc("Menu 28")
+        dtmf_enc.append(rset)
+
+        # DTMF - Decode
+        tmp = (str(_dtmfd.idcode)
+               .strip("\xFF")
+               )
+        rs = RadioSettingValueString(0, 16, tmp, False, CHARSET_NUMERIC)
+        rset = RadioSetting("dtmfd.idcode", "ID Code", rs)
+        rset.set_apply_callback(apply_dmtf_frame, _dtmfd.idcode, 10)
+        dtmf_dec.append(rset)
+
+        #
+        options = ["Off", "A", "B", "C", "D", "*", "#"]
+        rs = RadioSettingValueList(options, options[_dtmfd.grpcode])
+        rset = RadioSetting("dtmfd.grpcode", "Group Code", rs)
+        dtmf_dec.append(rset)
+
+        #
+        options = ["Off"] + ["%s" % x for x in range(1, 251)]
+        rs = RadioSettingValueList(options, options[_dtmfd.art])
+        rset = RadioSetting("dtmfd.art", "Auto Reset Time[s]", rs)
+        dtmf_dec.append(rset)
+
+        #
+        rs = RadioSettingValueBoolean(_settings.dani)
+        rset = RadioSetting("dani", "ANI", rs)
+        dtmf_dec.append(rset)
+
+        tmp = (str(_dtmfd.stuncode)
+               .strip("\xFF")
+               )
+        rs = RadioSettingValueString(0, 16, tmp, False, CHARSET_NUMERIC)
+        rset = RadioSetting("dtmfd.stuncode", "Stun Code", rs)
+        rset.set_apply_callback(apply_dmtf_frame, _dtmfd.stuncode, 10)
+        dtmf_dec.append(rset)
+
+        #
+        options = ["TX Inhibit", "TX/RX Inhibit"]
+        rs = RadioSettingValueList(options, options[_dtmfd.stuntype])
+        rset = RadioSetting("dtmfd.stuntype", "Stun Type", rs)
+        dtmf_dec.append(rset)
+
+        # DTMF - Autodial Memory
+        codes = self._memobj.dtmf_codes
+        i = 1
+        for dtmfcode in codes:
```

Comment:
Protip life hack for you. Whenever you need to iterate a list *and* have the index, you can do this:
```
for i, dtmfcode in enumerate(codes):
```
Now, `i` will be zero-origin isntead of 1 like you're doing here, but  it looks like you're using it as `i-1` anyway.

Reply:
Thanks for the tip. Appreciated.